### PR TITLE
CONVERT_TZ should return NULL if its arguments are invalid. (#11161)

### PR DESCRIPTION
**Pull request by tcmichael**
### What problem does this PR solve? 
CONVERT_TZ should return NULL if its arguments are invalid
Closes https://github.com/pingcap/tidb/issues/11161

### What is changed and how it works?
1. `evalTime` swallows error and makes `isNull` true when the arguments are invalid.
2. validate that zone is not "".

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tcmichael** [comment]
@SunRunAway PTAL

File: expression/builtin_time_test.go [hunk 1]
```diff
@@ -2450,11 +2450,14 @@ func (s *testEvaluatorSuite) TestConvertTz(c *C) {
...
 		{"2004-01-01 12:00:00", "-00:00", "-12:88", true, ""},
-		{"2004-01-01 12:00:00", "+10:82", "GMT", false, ""},
+		{"2004-01-01 12:00:00", "+10:82", "GMT", true, ""},
 		{"2004-01-01 12:00:00", "+00:00", "GMT", true, ""},
 		{"2004-01-01 12:00:00", "GMT", "+00:00", true, ""},
 		{20040101, "+00:00", "+10:32", true, "2004-01-01 10:32:00"},
 		{3.14159, "+00:00", "+10:32", true, ""},
+		{"2004-01-01 12:00:00", "", "", true, ""},
+		{"2004-01-01 12:00:00", "a", "", true, ""},
+		{"2004-01-01 12:00:00", "0", "", true, ""},
```
Comment (SunRunAway) [inline expression/builtin_time_test.go]: I suggest adding some similar tests to cover `toTz`?

Comment (tcmichael) [reply expression/builtin_time_test.go]: good idea.

File: expression/builtin_time_test.go [hunk 2]
```diff
@@ -2450,11 +2450,17 @@ func (s *testEvaluatorSuite) TestConvertTz(c *C) {
...
 		{"2004-01-01 12:00:00", "+00:00", "GMT", true, ""},
 		{"2004-01-01 12:00:00", "GMT", "+00:00", true, ""},
 		{20040101, "+00:00", "+10:32", true, "2004-01-01 10:32:00"},
 		{3.14159, "+00:00", "+10:32", true, ""},
+		{"2004-01-01 12:00:00", "", "GMT", true, ""},
+		{"2004-01-01 12:00:00", "GMT", "", true, ""},
+		{"2004-01-01 12:00:00", "a", "GMT", true, ""},
+		{"2004-01-01 12:00:00", "0", "GMT", true, ""},
+		{"2004-01-01 12:00:00", "GMT", "a", true, ""},
+		{"2004-01-01 12:00:00", "GMT", "0", true, ""},
```
Comment (SunRunAway) [inline expression/builtin_time_test.go]: Sorry, could you add some other tests to cover the `NULL` value of arguments?

File: expression/builtin_time.go [hunk 3]
```diff
@@ -5088,21 +5088,22 @@ func (b *builtinConvertTzSig) Clone() builtinFunc {
 }
 
 // evalTime evals CONVERT_TZ(dt,from_tz,to_tz).
+// `CONVERT_TZ` function returns NULL if the arguments are invalid.
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_convert-tz
 func (b *builtinConvertTzSig) evalTime(row chunk.Row) (types.Time, bool, error) {
 	dt, isNull, err := b.args[0].EvalTime(b.ctx, row)
 	if isNull || err != nil {
-		return types.Time{}, true, err
+		return types.Time{}, true, nil
```
Comment (lzmhhh123) [inline expression/builtin_time.go]: I have no idea whether directly ignore the error is ok. For safety, shall we only check the error is "unknown time zone"?

Comment (tcmichael) [reply expression/builtin_time.go]: https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_convert-tz
> This function returns NULL if the arguments are invalid.

Other invalid arguments should also be ignored。

**Comment by lzmhhh123** [comment]
@tcmichael Please add some tests in https://github.com/pingcap/tidb/blob/0ba4d0b41b0234a8bd1d70282dfd1bb28b9f7a30/expression/integration_test.go#L1798
Then I will approve.

**Comment by tcmichael** [comment]
@lzmhhh123 PTAL
